app/models/manageiq/providers/redhat/infra_manager/refresh_parser.rb

```diff
@@ -334,9 +340,28 @@ def self.host_inv_to_network_hashes(inv, guest_device_uids)
   def self.host_inv_to_os_hash(inv, hostname)
     return nil if inv.nil?
 
-    result = {:name => hostname}
-    result[:product_name] = 'linux'
-    result
+    {
+      :name         => hostname,
+      :product_type => 'linux',
+      :product_name => extract_host_os_name(inv),
+      :version      => extract_host_os_full_version(inv[:os])
+    }
+  end
+
+  def self.extract_host_os_full_version(host_os)
```

Comment:
@masayag the call to this method from `extract_host_os_name` is guarded with some `nil` checks.  But, the call to this method from `host_inv_to_os_hash` is *not* guarded with any `nil` checks.

Can you double check that to make sure some `nil` checks aren't missing?